demos/sample_azure_iot_pnp/sample_azure_iot_pnp.c

```diff
@@ -0,0 +1,1051 @@
+ * @param[in,out] pulIothubDeviceIdLength  Length of deviceId
+ */
+    static uint32_t prvIoTHubInfoGet( NetworkCredentials_t * pXNetworkCredentials,
+                                      uint8_t ** ppucIothubHostname,
+                                      uint32_t * pulIothubHostnameLength,
+                                      uint8_t ** ppucIothubDeviceId,
+                                      uint32_t * pulIothubDeviceIdLength );
+
+#endif /* democonfigENABLE_DPS_SAMPLE */
+
+/**
+ * @brief The task used to demonstrate the Azure IoT Hub API.
+ *
+ * @param[in] pvParameters Parameters as passed at the time of task creation. Not
+ * used in this example.
+ */
+static void prvAzureDemoTask( void * pvParameters );
+
+/**
+ * @brief Connect to endpoint with reconnection retries.
+ *
+ * If connection fails, retry is attempted after a timeout.
+ * Timeout value will exponentially increase until maximum
+ * timeout value is reached or the number of attempts are exhausted.
+ *
+ * @param pcHostName Hostname of the endpoint to connect to.
+ * @param ulPort Endpoint port.
+ * @param pxNetworkCredentials Pointer to Network credentials.
+ * @param pxNetworkContext Point to Network context created.
+ * @return uint32_t The status of the final connection attempt.
+ */
+static uint32_t prvConnectToServerWithBackoffRetries( const char * pcHostName,
+                                                      uint32_t ulPort,
+                                                      NetworkCredentials_t * pxNetworkCredentials,
+                                                      NetworkContext_t * pxNetworkContext );
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Static buffer used to hold MQTT messages being sent and received.
+ */
+static uint8_t ucMQTTMessageBuffer[ democonfigNETWORK_BUFFER_SIZE ];
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Generate max min payload.
+ */
+static AzureIoTResult_t prvInvokeMaxMinCommand( AzureIoTJSONReader_t * pxReader,
+                                                AzureIoTJSONWriter_t * pxWriter )
+{
+    AzureIoTResult_t xResult;
+    uint32_t ulSinceTimeLength;
+
+    /* Get the start time */
+    if( ( xResult = AzureIoTJSONReader_NextToken( pxReader ) )
+        != eAzureIoTSuccess )
+    {
+        LogError( ( "Error getting next token: result 0x%08x", xResult ) );
+    }
+    else if( ( xResult = AzureIoTJSONReader_GetTokenString( pxReader,
+                                                            ucCommandStartTimeValueBuffer,
+                                                            sizeof( ucCommandStartTimeValueBuffer ),
+                                                            &ulSinceTimeLength ) )
+             != eAzureIoTSuccess )
+    {
+        LogError( ( "Error getting token string: result 0x%08x", xResult ) );
+    }
+    else if( ( xResult = AzureIoTJSONWriter_AppendBeginObject( pxWriter ) )
```

Comment:
Nit - given we have the `/* Get the start time */` above, should we have something with like `/* Build the response buffer */` once we start building it up?